valhalla/sif/dynamiccost.h

```diff
@@ -669,6 +669,16 @@ class DynamicCost {
 
   virtual Cost BSSCost() const;
 
+  /*
+   * Determine whether an edge is filtered due to traffic closure.
+   * @param  edgeid         GraphId of the opposing edge.
+   * @return  Returns true if the edge is filtered due to live traffic closure, false if not.
+   */
+  inline virtual bool FilterClosed(const baldr::DirectedEdge* edge,
```

Comment:
im not a fan of making this public especially bcause it has some hidden state in there making use of `filter_closures_`

what if we make a generic public function like this:

```c++
inline virtual bool IsClosed(const baldr::DirectedEdge* edge,
                                   const baldr::GraphTile* tile) const {
￼    return (flow_mask_ & baldr::kCurrentFlowMask) && tile->IsClosed(edge);
￼  }
```
and inside of the `Filter` function we call it with the boolean `filter_closures_` and inside of the allowed function we call it with the boolean `ignore_closures`.

Reply:
I don't see the `Allowed` function using `ignore_closures` currently. Looks like its used during a-star search? Not sure of its impact if we move the `ignore_closures` check there.

Even with that, the basic issue IMO, still stands i.e during search (specifically in `is_search_filter_triggered`), how do we know the state of `ignore_closures`? `filter_closures` is already appropriately set taking `ignore_closures` into account, so I thought it would be good to expose it indirectly via the `Filter` function. I don't think doing so exposes an internal state, but rather rather exposes an API that just uses it.

Not sure how making `IsClosed` public helps here, since we still need to access `filter_closures_` during search. Maybe I'm missing something.

Reply:
Actually, I don't need to make `FilterClosed` (or `IsClosed`) public. I guess I can call `costing.Filter(...)` inside `is_search_filter_triggered`:
```c++
if ((road_class > min_road_class || road_class < max_road_class) ||
     (search_filter.exclude_tunnel_ && edge->tunnel()) ||
     (search_filter.exclude_bridge_ && edge->bridge()) ||
     (search_filter.exclude_ramp_ && (edge->use() == Use::kRamp)) ||
     (search_filter.exclude_closures_ && costing.Filter(edge, tile))) {
      return true;
```

Here's how I think it will work:

1. `ignore_closures: false`, `exclude_closures: true` (both defaults): `Costing::Filter` filters candidates due to closures Search filter is never called for candidates that are already removed by `Costing::Filter`.
2. `ignore_closures: true`, `exclude_closures: true or false`: This sets `filter_closures_` to `false` and `Costing::Filter` does not filter any candidates due to closures. search filter is also not triggered, since `Costing::Filter` always returns `0.0/false`
3. `ignore_closures: false`, `exclude_closures: false`: This sets `filter_closures_` to `false` and `Costing::Filter` does not filter any candidates due to closures. search filter is also not triggered, since `exclude_closures` is `false`.